### PR TITLE
ROE-2580, ROE-2571 - Remove Save and Resume feature flags

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2580
https://companieshouse.atlassian.net/browse/ROE-2571

### Change description
Removing the use of 'Save and Resume' feature flags as they have been On for > 12 months:
* FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022
* FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME_07032023

** PACKAGE.JSON TO BE UPDATED TO USE MATCHING API-SDK-NODE CHANGES**

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/shared/common.resume.submission.controller.ts [hunk 1]
```diff
@@ -104,9 +102,9 @@ const setWebApplicationData = (session: Session, appData: ApplicationData, trans
   appData[Transactionkey] = transactionId;
   appData[OverseasEntityKey] = overseaEntityId;
 
-  if (Object.keys(appData[OverseasEntityDueDiligenceKey] as OverseasEntityDueDiligence).length) {
+  if (Object.keys(appData[OverseasEntityDueDiligenceKey] ?? {}).length) {
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/controllers/shared/common.resume.submission.controller.ts]: updated

File: src/controllers/update/remove.cannot.use.controller.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { logger } from "../../utils/logger";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
   try {
-    logger.debugRequest(req, `GET ${config.SIGN_OUT_PAGE}`);
+    logger.debugRequest(req, `GET ${config.REMOVE_CANNOT_USE_PAGE}`);
```
Comment (mwestacott) [inline src/controllers/update/remove.cannot.use.controller.ts:7]: This change looks like nothing to do with 'save and resume'. Is it correct? (I imagine this question cropped up previously on PR to feature branch....)

File: test/controllers/check.your.answers.controller.spec.ts [hunk 3]
```diff
@@ -1177,29 +1167,8 @@ describe("POST tests", () => {
     expect(resp.text).toEqual(`${FOUND_REDIRECT_TO} ${PAYMENT_LINK_JOURNEY}`);
   });
 
-  test(`redirect the ${CONFIRMATION_PAGE} page after a successful post from ${CHECK_YOUR_ANSWERS_PAGE} page`, async () => {
```
Comment (mwestacott) [inline test/controllers/check.your.answers.controller.spec.ts:1180]: This test still sounds relevant but I guess it was specific to 'save and resume' and the scenario is covered by another test? Same for a number of tests being removed below...

File: src/controllers/update/remove.cannot.use.controller.ts (hunk 2 above)
Comment (markpit) [reply src/controllers/update/remove.cannot.use.controller.ts:7]: I think that was something I came across when debugging the tests, i noticed the page name didn't match the controller (must have been copied and pasted). I can change it back if you want.

File: test/controllers/check.your.answers.controller.spec.ts (hunk 3 above)
Comment (markpit) [reply test/controllers/check.your.answers.controller.spec.ts:1180]: I'm not sure, i'll look into it.

File: src/controllers/update/remove.cannot.use.controller.ts (hunk 2 above)
Comment (mwestacott) [reply src/controllers/update/remove.cannot.use.controller.ts:7]: No, all good.

File: test/controllers/check.your.answers.controller.spec.ts (hunk 3 above)
Comment (markpit) [reply test/controllers/check.your.answers.controller.spec.ts:1180]: they were basically repeats of existing tests, once the save and resume flag branch of code was made default in the controller the tests became almost identical to those just above. The redirect page difference is only due to the individual test overriding the redirect page name returned from the mock.
I've added some extra 'expects' in to beef up a couple of these tests in this area.

**Review by dlloyd1-ch — APPROVED** [review]
LGTM

**Pull request merged**